### PR TITLE
ISSUES-10056 support identifier argument for MySQL Database engine

### LINKED ISSUE #10056: CREATE DATABASE ENGINE = MySQL does not work
**Describe the bug**
```
CREATE DATABASE conv_main ENGINE = MySQL('host:3306', conv_main, 'metrika', 'password')
```

```
2020.04.06 04:14:34.359167 [ 172773 ] {905160b8-3bed-4129-ac2a-cd435384de4d} <Debug> executeQuery: (from [::1]:51104) CREATE DATABASE conv_main ENGINE = MySQL('host:3306', conv_main, 'metrika', 'password')
2020.04.06 04:14:34.359259 [ 172773 ] {905160b8-3bed-4129-ac2a-cd435384de4d} <Trace> ContextAccess (default): Access granted: CREATE DATABASE ON conv_main.*
2020.04.06 04:14:34.359588 [ 172049 ] {} <Trace> BaseDaemon: Received signal 11
2020.04.06 04:14:34.359778 [ 175713 ] {} <Fatal> BaseDaemon: ########################################
2020.04.06 04:14:34.359848 [ 175713 ] {} <Fatal> BaseDaemon: (version 20.4.1.1) (from thread 172773) (query_id: 905160b8-3bed-4129-ac2a-cd435384de4d) Received signal Segmentation fault (11).
2020.04.06 04:14:34.359890 [ 175713 ] {} <Fatal> BaseDaemon: Address: 0x80 Access: read. Address not mapped to object.
2020.04.06 04:14:34.359914 [ 175713 ] {} <Fatal> BaseDaemon: Stack trace: 0xcf5a0b6 0xcf5aa9a 0xcf51792 0xcf521c7 0xd46bf83 0xd46ed6d 0x8ff64d4 0x8ff73ec 0xe353807 0xe353c7b 0x104a40ce 0x1049f67c 0x7f73737cd6db 0x7f7373ea4a4f
2020.04.06 04:14:34.360749 [ 175713 ] {} <Fatal> BaseDaemon: 3. /home/milovidov/ClickHouse/build_gcc9/../src/Core/Field.h:398: DB::DatabaseFactory::getImpl(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::ASTStorage const*, DB::Context&) @ 0xcf5a0b6 in /usr/bin/clickhouse
2020.04.06 04:14:34.361626 [ 175713 ] {} <Fatal> BaseDaemon: 4. /home/milovidov/ClickHouse/build_gcc9/../src/Databases/DatabaseFactory.cpp:50: DB::DatabaseFactory::get(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::ASTStorage const*, DB::Context&) @ 0xcf5aa9a in /usr/bin/clickhouse
2020.04.06 04:14:34.363393 [ 175713 ] {} <Fatal> BaseDaemon: 5. /home/milovidov/ClickHouse/build_gcc9/../contrib/libcxx/include/string:1426: DB::InterpreterCreateQuery::createDatabase(DB::ASTCreateQuery&) @ 0xcf51792 in /usr/bin/clickhouse
2020.04.06 04:14:34.365126 [ 175713 ] {} <Fatal> BaseDaemon: 6. /home/milovidov/ClickHouse/build_gcc9/../src/Interpreters/InterpreterCreateQuery.cpp:722: DB::InterpreterCreateQuery::execute() @ 0xcf521c7 in /usr/bin/clickhouse
2020.04.06 04:14:34.366016 [ 175713 ] {} <Fatal> BaseDaemon: 7. /home/milovidov/ClickHouse/build_gcc9/../src/Interpreters/executeQuery.cpp:333: DB::executeQueryImpl(char const*, char const*, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*, bool) @ 0xd46bf83 in /usr/bin/clickhouse
2020.04.06 04:14:34.366907 [ 175713 ] {} <Fatal> BaseDaemon: 8. /home/milovidov/ClickHouse/build_gcc9/../src/Interpreters/executeQuery.cpp:578: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, bool) @ 0xd46ed6d in /usr/bin/clickhouse
2020.04.06 04:14:34.367768 [ 175713 ] {} <Fatal> BaseDaemon: 9. /home/milovidov/ClickHouse/build_gcc9/../programs/server/TCPHandler.cpp:253: DB::TCPHandler::runImpl() @ 0x8ff64d4 in /usr/bin/clickhouse
2020.04.06 04:14:34.368766 [ 175713 ] {} <Fatal> BaseDaemon: 10. /home/milovidov/ClickHouse/build_gcc9/../programs/server/TCPHandler.cpp:1243: DB::TCPHandler::run() @ 0x8ff73ec in /usr/bin/clickhouse
2020.04.06 04:14:34.369547 [ 175713 ] {} <Fatal> BaseDaemon: 11. /home/milovidov/ClickHouse/build_gcc9/../contrib/poco/Net/src/TCPServerConnection.cpp:57: Poco::Net::TCPServerConnection::start() @ 0xe353807 in /usr/bin/clickhouse
2020.04.06 04:14:34.370394 [ 175713 ] {} <Fatal> BaseDaemon: 12. /home/milovidov/ClickHouse/build_gcc9/../contrib/libcxx/include/atomic:856: Poco::Net::TCPServerDispatcher::run() @ 0xe353c7b in /usr/bin/clickhouse
2020.04.06 04:14:34.371293 [ 175713 ] {} <Fatal> BaseDaemon: 13. /home/milovidov/ClickHouse/build_gcc9/../contrib/poco/Foundation/include/Poco/Mutex_POSIX.h:59: Poco::PooledThread::run() @ 0x104a40ce in /usr/bin/clickhouse
2020.04.06 04:14:34.372109 [ 175713 ] {} <Fatal> BaseDaemon: 14. /home/milovidov/ClickHouse/build_gcc9/../contrib/poco/Foundation/include/Poco/AutoPtr.h:223: Poco::ThreadImpl::runnableEntry(void*) @ 0x1049f67c in /usr/bin/clickhouse
2020.04.06 04:14:34.372149 [ 175713 ] {} <Fatal> BaseDaemon: 15. start_thread @ 0x76db in /lib/x86_64-linux-gnu/libpthread-2.27.so
2020.04.06 04:14:34.372267 [ 175713 ] {} <Fatal> BaseDaemon: 16. /build/glibc_2.27-3ubuntu1yandex1/glibc-2.27/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:97: clone @ 0x121a4f in /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.27.so
```

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

fixed #10056 

Changelog category:
- Bug Fix

Changelog entry:
Add some arguments check and support identifier arguments for MySQL Database Engine

### CODE REVIEW

**Comment by qoega** [comment]
Docs should be fixed as well:
https://clickhouse.tech/docs/en/engines/database_engines/mysql/#creating-a-database

**Comment by zhang2014** [comment]
Test failure irrelevant to the modification

File: docs/en/engines/database_engines/mysql.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ You cannot perform the following queries:
 
 ``` sql
 CREATE DATABASE [IF NOT EXISTS] db_name [ON CLUSTER cluster]
-ENGINE = MySQL('host:port', 'database', 'user', 'password')
+ENGINE = MySQL('host:port', ['database' | database], 'user', 'password')
```
Comment (qoega) [inline docs/en/engines/database_engines/mysql.md:22]: Actually now I can't understand why we allow both variants here and have test `test_bad_arguments_for_mysql_database_engine` later.

There is different behavior in Storage and TableFunction?
```sql
CREATE TABLE default.t1_remote_mysql AS mysql('mysql1:3306', 'test', 't1_mysql_local', root, 'clickhouse')
```

**Comment by alexey-milovidov** [comment]
All table functions:
- mysql
- odbc
- jdbc
- remote
- cluster

Should have consistent behaviour:
We should allow to database and table names without quotes.

The same for MySQL database engine.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
The code for extraction of identifier/string should be reused somehow.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Ok. I will also add a trivial functional test to check that the query
```
CREATE DATABASE conv_main ENGINE = MySQL('host:3306', conv_main, 'metrika', 'password')
```
does not segfault.

**Comment by zhang2014** [comment]
> Ok. I will also add a trivial functional test to check that the query

Could this possibly broken unbundled's test?

**Pull request merged**